pipelines/common/src/main/java/org/openmrs/analytics/ParquetUtil.java

```diff
@@ -222,27 +228,44 @@ public List<GenericRecord> generateRecords(Bundle bundle) {
 		}
 		for (BundleEntryComponent entry : bundle.getEntry()) {
 			Resource resource = entry.getResource();
-			// TODO: Check why Bunsen returns IndexedRecord instead of GenericRecord.
-			records.add(convertToAvro(resource));
+			GenericRecord record = convertToAvro(resource);
+			if (record != null) {
+				records.add(record);
+			}
 		}
 		return records;
 	}
 	
-	public void writeRecords(Bundle bundle) throws IOException {
-		if (bundle.getTotal() == 0) {
+	public void writeRecords(Bundle bundle, Set<String> resourceTypes) throws IOException {
+		if (bundle.getEntry() == null) {
 			return;
 		}
 		for (BundleEntryComponent entry : bundle.getEntry()) {
 			Resource resource = entry.getResource();
-			write(resource);
+			if (resourceTypes == null || resourceTypes.contains(resource.getResourceType().name())) {
+				write(resource);
+			}
 		}
 	}
 	
 	@VisibleForTesting
+	@Nullable
 	GenericRecord convertToAvro(Resource resource) {
+		// TODO update Bunsen and get rid of this R3 conversion. Note that currently the conversion
+		// can fail for some R4 resource types (and they should be excluded). Once this is done we
+		// should remove the requirement for specifying resourceTypes when running the pipeline.
 		org.hl7.fhir.dstu3.model.Resource r3Resource = org.hl7.fhir.convertors.VersionConvertor_30_40
 		        .convertResource(resource, true);
+		if (r3Resource == null) {
+			log.error("R3 conversion failed; ignoring resource with type " + resource.getResourceType().name());
```

Comment:
Are we OK with returning  `null` if `r3Resource` is `null`. I get why we return `null` for the `r3Resource.getResourceType()` case, but not this

Reply:
The general problem is that the R4->R3 conversion can fail for various reasons (e.g., a new resource). In all these cases we should return `null` and handle it outside. These issues should go away once we fix #285 which I am planning to do soon.